### PR TITLE
Migrate Reports to OpenAPI

### PR DESCRIPTION
**Pull request by spinillos**
It migrates Reports to OpenAPI.

Depends of: https://github.com/grafana/grafana-openapi-client-go/pull/49

### CODE REVIEW

File: internal/resources/grafana/resource_report.go [hunk 1]
```diff
@@ -317,102 +326,94 @@ func ReadReport(ctx context.Context, d *schema.ResourceData, meta interface{}) d
...
 		},
-		Formats: []string{reportFormatPDF},
+		Formats: []models.Type{reportFormatPDF},
+	}
+
+	id := int64(d.Get("dashboard_id").(int))
+	uid := d.Get("dashboard_uid").(string)
+	if uid == "" {
+		// It triggers the old way to generate reports
+		report.DashboardID = id
```
Comment (spinillos) [inline internal/resources/grafana/resource_report.go:383]: We aren't able to retrieve dashboards by id anymore. 

We need to set it in the deprecated `dashboard_id` to make it works.

File: internal/resources/grafana/resource_alerting_rule_group.go [hunk 2]
```diff
@@ -235,7 +235,8 @@ func deleteAlertRuleGroup(ctx context.Context, data *schema.ResourceData, meta i
 	group := resp.Payload
 
 	for _, r := range group.Rules {
-		_, err := client.Provisioning.DeleteAlertRule(r.UID)
+		params := provisioning.DeleteAlertRuleParams{UID: r.UID}
```
Comment (julienduchesne) [inline internal/resources/grafana/resource_alerting_rule_group.go]: If you rebase, you can get rid of this change and the go mod update!

Comment (spinillos) [reply internal/resources/grafana/resource_alerting_rule_group.go]: Yeah! I saw that in your PR 😂

**Review by julienduchesne — APPROVED** [review]
Nice! :shipit:

**Pull request merged**